crates/biome_js_analyze/src/lint/nursery/no_duplicated_spread_props.rs

```diff
@@ -0,0 +1,104 @@
+use biome_analyze::{
+    Ast, Rule, RuleDiagnostic, RuleDomain, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_js_syntax::AnyJsxAttribute;
+use biome_js_syntax::{JsxAttributeList, JsxOpeningElement, JsxSelfClosingElement};
+use biome_rowan::{AstNode, declare_node_union};
+use biome_rule_options::no_duplicated_spread_props::NoDuplicatedSpreadPropsOptions;
+use std::collections::HashSet;
+
+declare_lint_rule! {
+    /// Disallow JSX prop spreading the same identifier multiple times.
+    ///
+    /// Enforces that any unique expression is only spread once.
+    /// Generally spreading the same expression twice is an indicator of a mistake since any attribute between the spreads may be overridden when the intent was not to.
+    /// Even when that is not the case this will lead to unnecessary computations being performed.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <div {...props} something="else" {...props} />
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <div something="else" {...props} />
+    /// ```
+    ///
+    pub NoDuplicatedSpreadProps {
+        version: "next",
+        name: "noDuplicatedSpreadProps",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintReact("jsx-props-no-spread-multi").same()],
+        domains: &[RuleDomain::React, RuleDomain::Solid],
+    }
+}
+
+declare_node_union! {
+    pub NoDuplicatedSpreadPropsQuery =
+        JsxOpeningElement
+        | JsxSelfClosingElement
+}
+
+fn validate_attributes(list: &JsxAttributeList) -> Option<String> {
+    let mut seen_spreads = HashSet::new();
+
+    for attribute in list {
+        if let AnyJsxAttribute::JsxSpreadAttribute(spread) = attribute
+            && let Some(argument) = spread.argument().ok()
+        {
+            let express = argument.as_js_identifier_expression()?;
+            let name = express.name().ok()?;
+            let value_token = name.value_token().ok()?;
+            let text = value_token.text_trimmed().to_string();
+            if !seen_spreads.insert(text.clone()) {
+                return Some(text);
+            }
+        }
+    }
+
+    None
+}
```

Comment:
As a general rule, we tend to have the implementation of the rule after the `impl Rule ` trait block